src/resolver.js

```diff
@@ -0,0 +1,108 @@
+const dns = require("dns");
+const isValidDomain = require("is-valid-domain");
+
+const logger = require("./logger");
+
+const dnslinkNamespace = "skynet-ns";
+const sponsorNamespace = "skynet-sponsor-key";
+const dnslinkRegExp = new RegExp(`^dnslink=/${dnslinkNamespace}/.+$`);
+const sponsorRegExp = new RegExp(`^${sponsorNamespace}=[a-zA-Z0-9]+$`);
+const dnslinkSkylinkRegExp = new RegExp(`^dnslink=/${dnslinkNamespace}/([a-zA-Z0-9_-]{46}|[a-z0-9]{55})`);
+const hint = `valid example: dnslink=/${dnslinkNamespace}/3ACpC9Umme41zlWUgMQh1fw0sNwgWwyfDDhRQ9Sppz9hjQ`;
+
+class Resolver {
+  async resolve(domainName) {
+    const lookup = `_dnslink.${domainName}`;
+
+    return new Promise((resolve) => {
+      dns.resolveTxt(lookup, (error, addresses) => {
```

Comment:
Ideally all these if conditions have comments explaining why we are checking for them.

The why and the context is important for others and reviewers to be able to onramp quickly.